bot.py

```diff
@@ -127,49 +217,95 @@ async def _subscribe(interaction: discord.Interaction, timeblock: Timeblock):
             name=Timeblock.Saturday.name, value=Timeblock.Saturday.value
         ),
         app_commands.Choice(name=Timeblock.Sunday.name, value=Timeblock.Sunday.value),
-    ]
+    ],
+    subscription=[
+        app_commands.Choice(name="Pair Programming", value="Programming"),
+        app_commands.Choice(name="Leetcode", value="Leetcode"),
+        app_commands.Choice(name="All", value="All"),
+    ],
 )
-async def _unsubscribe(interaction: discord.Interaction, timeblock: Timeblock):
+async def _unsubscribe(
+    interaction: discord.Interaction,
+    timeblock: Timeblock,
+    subscription: str = "All",
```

Comment:
This has to be `Optional[str]` or Discord won't treat it as truly optional, I think.